cli/src/commands/tui/events/network.rs

```diff
@@ -397,46 +405,52 @@ async fn get_gcloud_logs(app: Arc<Mutex<App>>, wk_client: &mut WKClient) -> Resu
                     )?;
                     let resource_names = vec![format!("projects/{}", cluster.google_project_id)];
 
-                    let mut log = match fetch_log_entries(
-                        Some(resource_names.clone()),
-                        Some(500),
-                        Some(filter.clone()),
-                        None,
-                        gcloud_access_token.clone(),
-                        wk_client,
-                    )
-                    .await
-                    {
-                        Ok(data) => data,
-                        Err(error) => {
-                            let mut app_ref = app.lock().await;
-                            app_ref.state.log_entries_error = Some(format!("{error}"));
-                            return Err(error);
+                    let log = if tailing {
+                        match fetch_log_entries(
+                            Some(resource_names.clone()),
+                            Some(1000),
+                            Some(filter.clone()),
+                            None,
+                            None,
+                            gcloud_access_token.clone(),
+                            wk_client,
+                        )
+                        .await
+                        {
+                            Ok(data) => data,
+                            Err(error) => {
+                                let mut app_ref = app.lock().await;
+                                app_ref.state.log_entries_error = Some(format!("{error}"));
+                                return Err(error);
+                            }
                         }
-                    };
-
-                    let mut next_page_token = log.next_page_token.clone();
-                    update_logs_entries(Arc::clone(&app), log.entries, &id).await;
-
-                    // repeat fetching logs until there is no next_page_token
-                    let app = Arc::clone(&app);
-                    while next_page_token.is_some() && next_page_token != Some("".to_string()) {
-                        log = fetch_log_entries(
+                    } else {
+                        let mut log = match fetch_log_entries(
```

Comment:
I suggest simplifying the code by consolidating it into a single `fetch_log_entries` function. We can modify the parameters based on tailing to enhance readability.

```suggestion
                    let mut log = match fetch_log_entries(
                        Some(resource_names.clone()),
                        Some(max_entries),
                        Some(filter.clone()),
                        None,
                        order_by,
                        gcloud_access_token.clone(),
                        wk_client,
                    )
                    .await
                    {
                        Ok(data) => data,
                        Err(error) => {
                            let mut app_ref = app.lock().await;
                            app_ref.state.log_entries_error = Some(format!("{error}"));
                            return Err(error);
                        }
                    };

                    if tailing {
                        log.entries = log
                            .entries
                            .map(|entries| entries.into_iter().rev().collect::<Vec<LogEntry>>());
                    }

```